### PR TITLE
Add tests for most of the library

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Adds a lot of drafts for the tests we always should've had.

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
Note that coverage reports are easy to obtain using `deno test --coverage ./test/`, however, they analyse the context and the API class and we would ideally like to omit these files as testing them is pointless.

**Comment by Satont** [comment]
Can you add gh action on any push for runing tests?

**Comment by Satont** [comment]
Also `deno task` command much be appreciated.

**Comment by KnorpelSenf** [comment]
> Can you add gh action on any push for runing tests?

Tests have always been running in CI on all three platforms since the early days of grammY.

> Also `deno task` command be appreciated.

Sure, I'll add it.

**Comment by KnorpelSenf** [comment]
`deno task` scripts are added and `npm run` scripts are updated. Can you review?

**Comment by KnorpelSenf** [comment]
@Satont are you aware of any code coverage reporters for LCOV files that work in GitHub Actions and that actually work? Neither romeovs/lcov-reporter-action nor mszlgr/lcov-reporter-action nor andybelltree/lcov-reporter-action are able to report test coverage for all tested files and without crashing, for a number of reasons.

**Comment by Satont** [comment]
> @Satont are you aware of any code coverage reporters for LCOV files that work in GitHub Actions and that actually work? Neither `romeovs/lcov-reporter-action` nor `mszlgr/lcov-reporter-action` nor `andybelltree/lcov-reporter-action` are able to report test coverage for all tested files and without crashing, for a number of reasons.

Codecov. They have integration with gh actions. Also check https://v1rtl.site/blog/coverage-in-deno, might be helpful.

**Comment by KnorpelSenf** [comment]
Thank you! Will check it out!

**Comment by KnorpelSenf** [comment]
@Satont thanks, this looks great! If you approve, we can merge and continue working on the test coverage in future PQs.

File: deno.jsonc [hunk 1]
```diff
@@ -1,4 +1,11 @@
 {
+    "tasks": {
+        "setup": "npm install && deno cache --check src/mod.ts",
+        "backport": "npm run backport",
+        "test": "deno test --seed=123456 ./test/",
+        "coverage": "deno task test --coverage=./test/cov_profile && deno coverage --lcov --output=./coverage.lcov ./test/cov_profile",
+        "bundle": "cd bundling && ./bundle-all.sh"
```
Comment (roj1512) [inline deno.jsonc:7]: Some of these tasks are duplicated in `package.json`. Maybe they should be removed either here or there.

Comment (KnorpelSenf) [reply deno.jsonc:7]: Thanks, fixed it up.

Comment (Satont) [reply deno.jsonc:7]: I guess he means you are duplciating scripts in package.json and deno.jsonc. It should be only in one place. So, you are not fixed the take.

**Review by Satont — APPROVED** [review]
Fix duplicates and then LGTM. 👍🏻

File: .github/workflows/deno.yml [hunk 2]
```diff
@@ -56,10 +56,30 @@ jobs:
...
+              with:
+                  deno-version: v1.x
+
+            - name: Create coverage files
+              run: deno task coverage
+
+            - name: Collect coverage
+              uses: codecov/codecov-action@v1.0.10 # upload the report on Codecov
+              with:
+                  file: ./coverage.lcov
```
Comment (roj1512) [inline .github/workflows/deno.yml:85]: I don't know if it is better or affects the speed, but you can collect the coverage, too, with one `checkout` and `setup-deno`, both in a single job.

File: deno.jsonc (hunk 1 above)
Comment (KnorpelSenf) [reply deno.jsonc:7]: Well, the logic is not duplicated. Having npm scripts for everything is mostly a convenience thing for Node users. Should we remove all npm scripts completely?

File: .github/workflows/deno.yml (hunk 2 above)
Comment (KnorpelSenf) [reply .github/workflows/deno.yml:85]: I'm not sure what you mean, how can a single job perform both the checkout and the Deno installation?

Comment (roj1512) [reply .github/workflows/deno.yml:85]: I mean doing the coverage collection in the `test` job, but that seems it is going to be not helpful as it is run 3 times and producing the same result.

**Pull request merged**